### ISSUE TITLE
inetmanet does not compile (error in voipstream application)

### ISSUE THREAD

**Issue by mfrey** [aarizaq/inetmanet-3.x#4]
Hi,

I'm trying to compile INETMANET 2.0 from the command line. I'm getting the following error:

<pre>
applications/voipstream/AudioOutFile.cc: In member function ‘void AudioOutFile::addAudioStream(AVCodecID, int, short int)’:
applications/voipstream/AudioOutFile.cc:46:21: error: ‘SAMPLE_FMT_S16’ was not declared in this scope
applications/voipstream/AudioOutFile.cc: In member function ‘void AudioOutFile::open(const char*, int, short int)’:
applications/voipstream/AudioOutFile.cc:83:35: error: ‘av_set_parameters’ was not declared in this scope
applications/voipstream/AudioOutFile.cc:118:44: error: ‘URL_WRONLY’ was not declared in this scope
applications/voipstream/AudioOutFile.cc: In member function ‘void AudioOutFile::write(void*, int)’:
applications/voipstream/AudioOutFile.cc:141:76: error: ‘av_get_bits_per_sample_format’ was not declared in this scope
applications/voipstream/AudioOutFile.cc:148:16: warning: ‘int avcodec_encode_audio(AVCodecContext*, uint8_t*, int, const short int*)’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:4209) [-Wdeprecated-declarations]
applications/voipstream/AudioOutFile.cc:148:76: warning: ‘int avcodec_encode_audio(AVCodecContext*, uint8_t*, int, const short int*)’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:4209) [-Wdeprecated-declarations]
make[1]: *** [../out/gcc-debug//src/applications/voipstream/AudioOutFile.o] Error 1
</pre>

From the search results I've got regarding the SAMPLE_FMT_S16 constant, it looks like that my version of ffmpeg might be too "new"? 

Thanks in advance!

 Best regards,
  Michael

**Comment by aarizaq** [comment]
You need install the VOIP libraries or disable the in feature VOIP

Project-propierties->OMNeT++->project features

Disable VoIPSTream, VoIPSTream examples

Apply

Ok

Rebuild

De: Michael Frey [mailto:notifications@github.com] 
Enviado el: lunes, 03 de diciembre de 2012 13:01
Para: aarizaq/inetmanet-2.0
Asunto: [inetmanet-2.0] inetmanet does not compile (error in voipstream application) (#4)

Hi,

I'm trying to compile INETMANET 2.0 from the command line. I'm getting the following error:

applications/voipstream/AudioOutFile.cc: In member function ���void AudioOutFile::addAudioStream(AVCodecID, int, short int)���:
applications/voipstream/AudioOutFile.cc:46:21: error: ���SAMPLE_FMT_S16��� was not declared in this scope
applications/voipstream/AudioOutFile.cc: In member function ���void AudioOutFile::open(const char_, int, short int)���:
applications/voipstream/AudioOutFile.cc:83:35: error: ���av_set_parameters��� was not declared in this scope
applications/voipstream/AudioOutFile.cc:118:44: error: ���URL_WRONLY��� was not declared in this scope
applications/voipstream/AudioOutFile.cc: In member function ���void AudioOutFile::write(void_, int)���:
applications/voipstream/AudioOutFile.cc:141:76: error: ���av_get_bits_per_sample_format��� was not declared in this scope
applications/voipstream/AudioOutFile.cc:148:16: warning: ���int avcodec_encode_audio(AVCodecContext_, uint8_t_, int, const short int_)��� is deprecated (declared at /usr/include/libavcodec/avcodec.h:4209) [-Wdeprecated-declarations]
applications/voipstream/AudioOutFile.cc:148:76: warning: ���int avcodec_encode_audio(AVCodecContext_, uint8_t_, int, const short int_)��� is deprecated (declared at /usr/include/libavcodec/avcodec.h:4209) [-Wdeprecated-declarations]
make[1]: **\* [../out/gcc-debug//src/applications/voipstream/AudioOutFile.o] Error 1
make[1]: Leaving directory `/home/frey/Desktop/Projekte/simulation/inetmanet-2.0/src'
make: **\* [all] Error 2

From the search results I've got regarding the SAMPLE_FMT_S16 constant, it looks like that my version of ffmpeg might be too "new"? 

Thanks in advance!

Best regards,
Michael

—
Reply to this email directly or view it on GitHub https://github.com/aarizaq/inetmanet-2.0/issues/4 . 

Imagen quitada por el remitente.

**Comment by mfrey** [comment]
What are the "VOIP libraries"? The libraries listed in the README of the voipstream application (libavcodec-dev libavformat-dev libavutil-dev) are installed on the system. I've tried it on two different machines with two different linux distributions and the error is the same.  In addition to the libraries I've also installed ffmpeg. However, the error remains. I've checked the code and there is a define '#ifndef HAVE_FFMPEG', but I haven't looked yet where it is set. Any comments?
